src/tensor_outputs/XDMFTensorOutput.C

```diff
@@ -333,6 +337,11 @@ XDMFTensorOutput::output()
           data.append_attribute("Precision") = "8";
           raw_size *= 8;
         }
+        else if (buffer.dtype() == torch::kInt32 || buffer.dtype() == torch::kInt64)
+        {
+          data.append_attribute("Precision") = "1";
+          raw_size *= 1;
+        }
```

Comment:
wait... isn't the raw size for `torch::kInt32` 4 (32bit) and for `torch::kInt64` 8? This would result in truncated writes